Set/ViewController.swift

```diff
@@ -107,67 +268,122 @@ class ViewController: UIViewController {
             button.isEnabled = true
             needToEndGame = false
         }
+        
+        computerMatchedButtons = [UIButton]()
+        computerSelectedButtons = [UIButton]()
         game = SetGame()
+        selectGameMode()
         initGameBoard()
         freeButtonIndex = game.numOfCardsOnStart
+        computerSelectedButtons = [UIButton]()
         selectedButtons = [UIButton]()
+        computerMatchedButtons = [UIButton]()
         matchedButtons = [UIButton]()
         needToDealNewCards = false
         dealCard.isEnabled = true //enable deal 3 cards button
         
         updateUI()
     }
     
+    @IBOutlet weak var computerScore: UILabel!
+    
+    @IBOutlet weak var computerStatusIndicator: UILabel!
+    
     @IBOutlet weak var dealCard: UIButton!
     
     @IBAction func dealCardsAndAddToGameBoard(_ sender: UIButton) {
-       checkIfNeedToEnd()
-        
+        checkIfNeedToEnd()
+        dealCardsAndAddToGameBoard()
+        playerMadeMove = false
+
+        if game.cardsOnGameBoard.count == game.maxGameBoardCapacity { // insufficient cards in the game board. The button is disabled
+            sender.isEnabled = false
+        }
+        needToDealNewCards = false
+//        matchedButtons.removeAll()
+//        computerMatchedButtons.removeAll()
+        updateUI()
+    }
+    
+    func dealCardsAndAddToGameBoard() {
         let threeNewCards = game.dealThreeNewCards() // get three new cards from the deck
         if threeNewCards.count == 3 {
-            if matchedButtons.count == 3 { // replace 3 matched cards with 3 new ones from the deck
-                let threeOldIndexes = [matchedButtons[0].tag, matchedButtons[1].tag, matchedButtons[2].tag]
-                for matchIndex in 0..<3 { // find the required button in buttons array
-                    for buttonIndex in 0..<buttons.count {
-                        if buttons[buttonIndex].tag == matchedButtons[matchIndex].tag { // we are on the right button
-                            let shade = printShape(ofShape: shapeToShading[shapes[threeNewCards[matchIndex].shape]]![threeNewCards[matchIndex].shading].string, times: threeNewCards[matchIndex].numOfShapes + 1)
-                            let attString = NSAttributedString(string: shade, attributes: [NSAttributedStringKey.foregroundColor : getColor(forCard: threeNewCards[matchIndex])])
-                            buttons.setButton(atIndex: buttonIndex, tag: threeNewCards[matchIndex].identifier, attributedString: attString, for: UIControlState.normal)
-                            changeShape(ofButton: buttons[buttonIndex])
-                        }
-                    }
-                }
-                addThreeNewCardsToSamePlaces(threeOldCardsPlaces: threeOldIndexes, threeCardsToAdd: threeNewCards)
+            playerMadeMove = true
```

Comment:
if the computer can call `dealCardsAndAddToGameBoard` - how is it true that `playerMadeMove` ?